strategies/failed/src/main/java/org/arquillian/smart/testing/strategies/failed/InProjectTestReporterLoader.java

```diff
@@ -38,11 +47,12 @@
                         }
                     })
                     .map(is -> {
-                        try {
-                            return SurefireReader.loadTestResults(is);
-                        } catch (XMLStreamException e) {
-                            return null;
+                        final Optional<TestResultParser> testResultParser = javaSPILoader.onlyOne(TestResultParser.class, trp -> "junit".equals(trp.type()));
+                        if (!testResultParser.isPresent()) {
+                            throw new IllegalArgumentException("No JUnit Test Result Parser found in classpath");
```

Comment:
If we use SPI it doesn't necessary mean it is "JUnit Test Result Parser", or does it?